src/main/java/io/neonbee/internal/verticle/ServerVerticle.java

```diff
@@ -76,57 +76,15 @@ public void handle(RoutingContext routingContext) {
-            rootRoute.handler(new CorrelationIdHandler(config.getCorrelationStrategy()));
-            rootRoute.handler(
-                    TimeoutHandler.create(SECONDS.toMillis(config.getTimeout()), config.getTimeoutStatusCode()));
-            rootRoute.handler(new CacheControlHandler());
-            rootRoute.handler(new InstanceInfoHandler());
-
-            createSessionStore(vertx, config.getSessionHandling()).map(SessionHandler::create)
-                    .ifPresent(sessionHandler -> rootRoute
-                            .handler(sessionHandler.setSessionCookieName(config.getSessionCookieName())));
-
-            // add all endpoint handlers as sub-routes here
-            mountEndpoints(router, config.getEndpointConfigs(), createAuthChainHandler(config.getAuthChainConfig()),
-                    new HooksHandler()).onFailure(startPromise::fail).onSuccess(nothing -> {
-                        // the NotFoundHandler fails the routing context finally
-                        router.route().handler(new NotFoundHandler());
-
-                        // Use the port passed via command line options, instead the configured one.
-                        Optional.ofNullable(NeonBee.get(vertx).getOptions().getServerPort()).ifPresent(config::setPort);
-
-                        vertx.createHttpServer(config /* ServerConfig is a HttpServerOptions subclass */)
-                                .exceptionHandler(throwable -> {
-                                    LOGGER.error("HTTP Socket Exception", throwable);
-                                }).requestHandler(router).listen().onSuccess(httpServer -> {
-                                    this.httpServer = httpServer;
-                                    if (LOGGER.isInfoEnabled()) {
-                                        LOGGER.info("HTTP server started on port {}", httpServer.actualPort());
-                                    }
-                                    if (LOGGER.isDebugEnabled()) {
-                                        LOGGER.debug("HTTP server configured with routes: {}", router.getRoutes()
-                                                .stream().map(Route::toString).collect(Collectors.joining(",")));
-                                    }
-                                }).onFailure(cause -> {
-                                    LOGGER.error("HTTP server could not be started", cause);
-                                }).<Void>mapEmpty().onComplete(startPromise);
-                    });
-        } catch (Exception e) {
-            LOGGER.error("Server could not be started", e);
-            startPromise.fail(e);
-        }
+        createRouter(config).compose(router -> {
+            Optional<AuthenticationHandler> ach = createAuthChainHandler(config.getAuthChainConfig());
+            return mountEndpoints(router, config.getEndpointConfigs(), ach, new HooksHandler()).map(v -> {
```

Comment:
Instead of using map(v->) use onSuccess

Reply:
I used map to ensure that the handler is added before createHttpServer is called.  Would it be a problem in case that the handler is added afterwards?

Reply:
I've tested it, it doesn't matter. Will use onSuccess